modules/core/src/lib/deck-picker.ts

```diff
@@ -361,14 +363,17 @@ export default class DeckPicker {
 
     // Convert from canvas top-left to WebGL bottom-left coordinates
     // And compensate for pixelRatio
+    // @ts-expect-error TODO - assuming WebGL context
     const pixelRatio = this.device.canvasContext.cssToDeviceRatio();
```

Comment:
Hmm, the optionality of the canvasContext clearly causes a lot of headache for the deck. Wonder if we should add a `device.getCanvasContext()` that throws instead of returns `null` so app doesn't need to be sprinkled with these ts-ignores.

Or maybe a dummy canvas context that can provide default values for these pixel queries..

It could be done in the app but it is not elegant as it would need to be repeated in many place.s
```suggestion
    const pixelRatio = this.device.canvasContext?.cssToDeviceRatio?.() || 1;
```

Reply:
I do not like optional chaining unless we know it's actually going to catch something. If you look at transpiled code it's incredibly verbose.

Reply:
Do we need to transpile it? https://caniuse.com/mdn-javascript_operators_optional_chaining

Reply:
But in this particular case the app also has to come up with its own fallback so I agree that it is not a good choice.